pinot-common/src/main/java/org/apache/pinot/common/request/context/RequestContextUtils.java

```diff
@@ -352,10 +352,10 @@ public static FilterContext getFilter(FunctionContext filterFunction) {
   }
 
   private static String getStringValue(ExpressionContext expressionContext) {
-    if (expressionContext.getType() != ExpressionContext.Type.LITERAL) {
+    if(expressionContext.getType() != ExpressionContext.Type.LITERAL){
       throw new BadQueryRequestException(
           "Pinot does not support column or function on the right-hand side of the predicate");
     }
-    return expressionContext.getLiteral();
+    return expressionContext.getLiteralString();
   }
```

Comment:
add a new function
```
getLiteralContext(ExpressionContext expressionContext) { 
  // ...
}
```

and make 
```
@Deprecated
getStringValue() {
  LiteralContext literalContext = getLiteralContext(...);
  return literalContext.getDataType().convertToString(literalContext.getValue());
}
```
later we make all the usage of getStringValue() go to getLiteralContext

Reply:
We can do this when we deprecate getLiteralString?